### ISSUE TITLE
Istio fails to install after a Helm delete

### ISSUE THREAD

**Issue by jpreese** [istio/istio#13248]
**Describe the bug**
ValidatingWebhookConfiguration resource is created by Galley. This prevents helm from knowing about the resource, and ends up dangling on a helm delete.

If ValidatingWebhookConfiguration exists prior to helm install, the installation will fail because it is then assumed Galley is up and running, but it may not be.

**Expected behavior**
Explicitly define the ValidatingWebhookConfiguration resource so it can be tracked.

**Steps to reproduce the bug**
```code
On a fresh Kubernetes cluster (using steps from https://istio.io/docs/setup/kubernetes/install/helm/)

// helm/tiller setup
kubectl apply -f install/kubernetes/helm/helm-service-account.yaml
helm init --service-account tiller

// istio install via helm
helm install install/kubernetes/helm/istio-init --name istio-init --namespace istio-system
helm install install/kubernetes/helm/istio --name istio --namespace istio-system

// delete istio release
helm delete --purge istio

// attempt to install again
helm install install/kubernetes/helm/istio --name istio --namespace istio-system
```

**Version**
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:38:32Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:25:46Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

Istio 1.1.2

**Environment**
Docker Desktop for Windows

**Comment by sdake** [comment]
@jpreese thank you for the report.  I am pretty sure we test this scenario, but we will get back to you with results either way.

Cheers
-steve

**Comment by jpreese** [comment]
Sounds good! It's not a huge deal right now. We're just explicitly deleting the hook after an uninstall.

**Comment by costinm** [comment]
We should make it consistent with the webhook injector - the main issue is ordering, if the webhook is installed we can't create Istio CRDs until galley is up, and the CA is tricky to setup.

**Comment by sdake** [comment]
@jpreese regarding this issue, I think the project has a fairly deep understanding of the problem here.  I'm not clear if we have a solution yet and this likely won't be solved in 1.3 for some of the reasons @costinm indicated.

**Comment by jpreese** [comment]
No worries from this side! We'll most likely migrate to the Kustomized distribution of Istio as it seems to be a much better fit. Worst case, we can always run a cleanup after a teardown.

**Comment by ayj** [comment]
The validatingwebhookconfiguration's ownerReference points to the galley deployment. k8s typically GCs the webhook config when the deployment is deleted. 

Here's an alternative:
1) add the validatingwebhookconfiguration back to the top-level manifest. 
2) change the default failurePolicy to IGNORE
3) update galley's webhook config reconciliation logic to only patch the `caBundle` *and* change the failurePolicy to FAIL. This restricted reconciliation step only occurs after Galley is ready to serve requests, similar to the current implementation.

(1) allows helm / installer to track the config as any other k8s resource
(2) allows CR to be created even when Galley isn't available (https://github.com/istio/istio/issues/13248#issuecomment-513329919). 
(3) retains the logic to only "enable" the fail-closed webhook policy when Galley is ready to serve requests.

(2) needs more testing. I'm not what latency it introduces to the install if every apiserver has to timeout on a not-ready webhook.

**Comment by howardjohn** [comment]
@ayj is this fixed by #16857?

**Comment by ayj** [comment]
No. The validatingwebhookconfiguration is owned by the `istio-system` namespace but that isn't deleted by `helm delete`. Users still need to manually delete the namespace.

**Comment by jpreese** [comment]
Its been quite some time since I last looked, but I don't remember that being the case. Even after deleting istio-system, the webhook was still in the cluster.

https://istio.io/docs/ops/setup/validation/

**Comment by ayj** [comment]
The namespace ownership fix went into master (post 1.3 release).

**Comment by howardjohn** [comment]
I think requiring to delete the ns is acceptable, maybe just want to  document here: https://istio.io/docs/setup/install/helm/#uninstall - it says to delete ns with `helm template` but not tiller
